### PR TITLE
C2PA-273: Use Table trait

### PR DESCRIPTION
**Pull request by vancejc-mt**
## Changes in this pull request
_Give a narrative description of what has been changed._

Slight change to make things a little DRYer (and more structured); introduced the `Table` trait (renaming the old `Table` enum to `NamedTable`) which holds the `len()`/`checksum()`/`write()` calls.  Our tables now implement this trait (using the functions they already had) and the `NamedTable` enumeration also implements this trait, allowing us to avoid doing the switch on `write()` later on (similar to how we do `checksum()` and `len()`).

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [ ] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

## Validation

- Pull this branch in this repo
- Clone the `c2patool` repo (in a sibling directory), using the `feature/C2PA-262/localSfntSupport` branch (this looks locally for the `c2pa-rs` repo in the same directory)
- Build `c2patool` (`cargo build`)
- Run the tool against a font using the same manifest json
  -  `cd target/debug`
  - `./c2patool.exe "C:\jira\c2pa307_pr\ALGER.TTF" -m C:\src\c2patool\sample\test.json -f -o "C:\jira\c2pa307_pr\ALGER_out.TTF"`

### CODE REVIEW

File: sdk/src/asset_handlers/sfnt_io.rs [hunk 1]
```diff
@@ -302,11 +302,17 @@ impl SfntFont {
             let offset: u64 = entry.offset as u64;
             let size: usize = entry.length as usize;
             // Create a table instance for it.
-            let table: Table = {
+            let table: NamedTable = {
                 match entry.tag {
-                    C2PA_TABLE_TAG => Table::C2PA(TableC2PA::from_reader(reader, offset, size)?),
-                    HEAD_TABLE_TAG => Table::Head(TableHead::from_reader(reader, offset, size)?),
-                    _ => Table::Unspecified(TableUnspecified::from_reader(reader, offset, size)?),
+                    C2PA_TABLE_TAG => {
```
Comment (brogdonm) [inline sdk/src/asset_handlers/sfnt_io.rs]: I would suggest DRYing this up as well and provide a helper method in a `impl NamedTable`, maybe like so:

```rust
impl NamedTable {
    /// Reads a new instance from the given source.
    pub(crate) fn from_reader<T: Read + Seek + ?Sized>(
        tag: &SfntTag,
        reader: &mut T,
        offset: u64,
        length: usize,
    ) -> Result<Self> {
        match *tag {
            C2PA_TABLE_TAG => Ok(NamedTable::C2PA(TableC2PA::from_reader(
                reader, offset, length,
            )?)),
            HEAD_TABLE_TAG => Ok(NamedTable::Head(TableHead::from_reader(
                reader, offset, length,
            )?)),
            _ => Ok(NamedTable::Unspecified(TableUnspecified::from_reader(
                reader, offset, length,
            )?)),
        }
    }
}
```

And then could clean up here and in woff with:

```rust
let table = NamedTable::from_reader(&entry.tag, reader, offset, size)?;
```

File: sdk/src/asset_handlers/woff_io.rs [hunk 2]
```diff
@@ -327,11 +327,17 @@ impl WoffFont {
             let offset: u64 = entry.offset as u64;
             let size: usize = entry.compLength as usize;
             // Create a table instance for it.
-            let table: Table = {
+            let table: NamedTable = {
```
Comment (brogdonm) [inline sdk/src/asset_handlers/woff_io.rs]: Same comment here as in the sfnt_io handler.

**Review by brogdonm — CHANGES_REQUESTED** [review]
Suggestion to DRY a little more.

File: sdk/src/asset_handlers/woff_io.rs (hunk 2 above)
Comment (vancejc-mt) [reply sdk/src/asset_handlers/woff_io.rs]: Done and done

File: sdk/src/asset_handlers/sfnt_io.rs (hunk 1 above)
Comment (vancejc-mt) [reply sdk/src/asset_handlers/sfnt_io.rs]: Done and done

**Review by brogdonm — APPROVED** [review]
Looks good!

**Pull request merged**